src/common/utils/fetch-client.ts

```diff
@@ -0,0 +1,107 @@
+import { FetchError } from './fetch-error';
+
+export class FetchClient {
+  constructor(readonly baseURL: string, readonly options?: RequestInit) {}
+
+  async get(
+    path: string,
+    options: { params?: Record<string, any>; headers?: HeadersInit },
+  ) {
+    const resourceURL = this.getResourceURL(path, options.params);
+    const response = await this.fetch(resourceURL, {
+      headers: options.headers,
+    });
+    return { data: await response.json() };
+  }
+
+  async post<Entity = any>(
+    path: string,
+    entity: Entity,
+    options: { params?: Record<string, any>; headers?: HeadersInit },
+  ) {
+    const resourceURL = this.getResourceURL(path, options.params);
+    const bodyIsSearchParams = entity instanceof URLSearchParams;
+    const contentTypeHeader = bodyIsSearchParams
+      ? { 'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8' }
+      : undefined;
+    const body = bodyIsSearchParams ? entity : JSON.stringify(entity);
```

Comment:
Support body as `URLSearchParams` for the one use-case we have for SSO. (`getProfileAndToken`)